content/docs/user-guide/pipelines/running-pipelines.md

```diff
@@ -108,6 +108,160 @@ stages:
+        data/test_data/
+        data/train_data/
+```
+
+We can modify the `evaluate` stage and DVC will only pull the necessary data to
+run that stage (`models/model.pkl` `data/test_data/`) while skipping the rest of
+the stages:
+
+```cli
+$ dvc exp run
+'data/pool_data.dvc' didn't change, skipping
+Stage 'data_split' didn't change, skipping
+Stage 'train' didn't change, skipping
+Running stage 'evaluate':
+...
+```
+
+## Verify Pipeline Status
+
+In scenarios like CI jobs, you may want to check that the pipeline is up to date
+without pulling or running anything. You can check that nothing has changed:
+
+<details>
+
+### Clean example
+
+In the example below, data is missing because nothing has been pulled, but
+otherwise the pipeline is up to date.
+
+```cli
+$ dvc status
+data_split:
+        changed deps:
+                deleted:            data/pool_data
+        changed outs:
+                not in cache:       data/test_data
+                not in cache:       data/train_data
+train:
+        changed deps:
+                deleted:            data/train_data
+        changed outs:
+                not in cache:       models/model.pkl
+evaluate:
+        changed deps:
+                deleted:            data/test_data
+                deleted:            models/model.pkl
+data/pool_data.dvc:
+        changed outs:
+                not in cache:       data/pool_data
+```
+
+</details>
+
+```cli
+$ dvc exp run --allow-missing --dry
+Reproducing experiment 'agley-nuke'
+'data/pool_data.dvc' didn't change, skipping
+Stage 'data_split' didn't change, skipping
+Stage 'train' didn't change, skipping
+Stage 'evaluate' didn't change, skipping
+```
+
+If anything is not up to date, the pipeline will fail:
+
+<details>
+
+### Dirty example
+
+In the example below, the `data_split` parameter in `params.yaml` was modified,
+so the pipeline is not up to date.
+
+```cli
+$ dvc status
+data_split:
+        changed deps:
+                deleted:            data/pool_data
+                params.yaml:
+                        modified:           data_split
+        changed outs:
+                not in cache:       data/test_data
+                not in cache:       data/train_data
+train:
+        changed deps:
+                deleted:            data/train_data
+        changed outs:
+                not in cache:       models/model.pkl
+evaluate:
+        changed deps:
+                deleted:            data/test_data
+                deleted:            models/model.pkl
+data/pool_data.dvc:
+        changed outs:
+                not in cache:       data/pool_data
+```
+
+</details>
+
+```cli
+$ dvc exp run --allow-missing --dry
+Reproducing experiment 'dozen-jogs'
+'data/pool_data.dvc' didn't change, skipping
+ERROR: failed to reproduce 'data_split': [Errno 2] No such file or directory: '/private/tmp/example-get-started-experiments/data/pool_data'
+```
+
+You can also check that all data exists on the remote. The command below will
```

Comment:
Why do we need this check?

Reply:
I had to update the command, but it's needed because `--allow-missing` won't actually check whether the missing data exists on the remote. However, it's probably important to users to ensure all data has been pushed so that the pipeline could be reproduced from scratch if needed. See the initial request for this feature in https://github.com/iterative/dvc/issues/5369.